apps/myaccount/src/api/consents.ts

```diff
@@ -89,6 +97,64 @@ export const fetchConsentReceipt = (receiptId: string): Promise<any> => {
         });
 };
 
+/**
+ * Fetches multiple {@link PurposeModel} by given a set of IDs. This function is useful
+ * when we need to get some detailed information about a consent receipts' purpose(s).
+ * This is because when the client use a method like {@link fetchConsentReceipt} it
+ * only gives the receipt with limited information.
+ *
+ * Context: -
+ * A consent can have many services, under services we have many purposes, and
+ * each purpose has many pii-category claims.
+ *
+ * Example: -
+ * Assume that we fetched a receipt {@link ConsentReceiptInterface} by its ID.
+ * Now we can easily access all of its properties. Let's say now you want to
+ * display all the piiCategories in each purpose regardless of whether the user
+ * granted or denied that claim.
+ *
+ * However, now you will run into a limitation where {@link ConsentReceiptInterface}
+ * only contains the granted claims in its: {@code receipt.services.each(purposes.
+ * each(purpose.piiCategory))} array. In this case you have to use this method to get
+ * detailed info about each of its purposes.
+ *
+ * Usage: -
+ * This service method will accept multiple {@code purposeIDs} and aggregate
+ * them to make concurrent requests for each ID. You can pass an {@code number[]}
+ * argument which contains only one purposeID as well.
+ *
+ * @param {Iterable[]} purposeIDs
+ * @return {PurposeModel[]} response data
+ */
+export const fetchPurposesByIDs = async (purposeIDs: Iterable<number>): Promise<PurposeModel[]> => {
+
+    const requestConfigurations: AxiosRequestConfig[] = [];
+    const url = store.getState().config.endpoints.consentManagement.purpose.getPurpose;
+
+    for (const purposeID of purposeIDs) {
+        const requestConfiguration: AxiosRequestConfig = {
+            headers: {
+                "Accept": "application/json",
+                "Content-Type": "application/json",
+                "Access-Control-Allow-Origin": store.getState().config.deployment.clientHost
```

Comment:
Let's remove `Access-Control-Allow-Origin` since it is not required. Refactor any similar once you find. Similar refactoring was done in the SDK in https://github.com/asgardio/asgardio-js-oidc-sdk/pull/63
```suggestion
```

Reply:
Applied this change in commit d5c7820